### PR TITLE
com.nvidia.spark.rapids.ColumnarRdd not exposed to user for XGBoost

### PR DESCRIPTION
**Pull request by tgravescs**
fixes https://github.com/NVIDIA/spark-rapids/issues/3601

we were not exposing ColumnarRDD as an unshimmed class.  This fixes that. I tested on xgboost with the python notebook and verified it works now.

This is a bit different since its an object and not a real class we instantiate, so if we have better ways to do this let me know, or we can followup as I'm sure we will have more things that need things like this.

### CODE REVIEW

**Comment by tgravescs** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/ColumnarRdd.scala [hunk 1]
```diff
@@ -39,11 +38,16 @@ import org.apache.spark.sql.rapids.execution.InternalColumnarRddConverter
...
+
   def apply(df: DataFrame): RDD[Table] = {
-    InternalColumnarRddConverter(df)
+    convert(df)
   }
 
   def convert(df: DataFrame): RDD[Table] = {
-    InternalColumnarRddConverter.convert(df)
+    internalConverter
+      .getDeclaredMethod("convert", classOf[DataFrame])
```
Comment (jlowe) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/ColumnarRdd.scala:50]: Shouldn't we be caching the method rather than looking it up each time?

**Review by revans2 — DISMISSED** [review]

**Comment by tgravescs** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/ColumnarRdd.scala (hunk 1 above)
Comment (tgravescs) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/ColumnarRdd.scala]: yes updated

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimLoader.scala [hunk 2]
```diff
@@ -289,6 +289,12 @@ object ShimLoader extends Logging {
     shimProviderClass = classname
   }
 
+  def newClassOf(className: String): Class[_] = {
```
Comment (gerashegalov) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimLoader.scala:292]: nit: consider `loadClass` as the name

nit be made resusable for newInstanceOf

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimLoader.scala [hunk 3]
```diff
@@ -343,4 +349,8 @@ object ShimLoader extends Logging {
   def newInternalExclusiveModeGpuDiscoveryPlugin(): ResourceDiscoveryPlugin = {
     newInstanceOf("com.nvidia.spark.rapids.InternalExclusiveModeGpuDiscoveryPlugin")
   }
+
+  def newColumnarRDDClass(): Class[_] = {
```
Comment (gerashegalov) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimLoader.scala:353]: nit: I'd name newXyz for methds returning objects and loadXyz for classes

**Comment by tgravescs** [comment]
build

**Review by gerashegalov — APPROVED** [review]
LGTM

**Review by jlowe — APPROVED** [review]

**Pull request merged**